pkg/api/dashboard_test.go

```diff
@@ -13,10 +14,58 @@ import (
 	"github.com/grafana/grafana/pkg/services/dashboards"
 	"github.com/grafana/grafana/pkg/services/provisioning"
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/require"
 
 	. "github.com/smartystreets/goconvey/convey"
 )
 
+func TestGetHomeDashboard(t *testing.T) {
+	req := &models.ReqContext{SignedInUser: &models.SignedInUser{}}
+	cfg := setting.NewCfg()
+	err := cfg.Load(&setting.CommandLineArgs{
+		HomePath: "../../",
+	})
+
+	require.NoError(t, err)
+	bus.AddHandler("test", func(query *models.GetPreferencesWithDefaultsQuery) error {
+		query.Result = &models.Preferences{
+			HomeDashboardId: 0,
+		}
+		return nil
+	})
+
+	tcs := []struct {
+		defaultSetting        string
+		expectedDashboardPath string
+	}{
+		{defaultSetting: "", expectedDashboardPath: "../../public/dashboards/home.json"},
+		{defaultSetting: "../../public/dashboards/default.json", expectedDashboardPath: "../../public/dashboards/default.json"},
+	}
+
+	for _, tc := range tcs {
+		dash := dtos.DashboardFullWithMeta{}
+		dash.Meta.IsHome = true
+		dash.Meta.CanEdit = req.SignedInUser.HasRole(models.ROLE_EDITOR)
+		dash.Meta.FolderTitle = "General"
+
+		homeDashJSON, err := ioutil.ReadFile(tc.expectedDashboardPath)
+		setting.DefaultHomeDashboardPath = tc.defaultSetting
+		require.NoError(t, err)
```

Comment:
Can you move this line to 52, so it's co-located with the originating statement?